### PR TITLE
SpiNNaker support

### PR DESCRIPTION
**Pull request by ej159**
This adds the necessary code to use SpiNNaker as a simulator.

### CODE REVIEW

**Comment by rbodo** [comment]
Fantastic, Ed, thanks!

I'll try to review and test this PR within the next two weeks.

File: snntoolbox/parsing/utils.py [hunk 1]
```diff
@@ -457,7 +507,8 @@ def try_insert_flatten(self, layer, idx, name_map):
         previous_layers = self.get_inbound_layers(layer)
         prev_layer_output_shape = self.get_output_shape(previous_layers[0])
         if len(output_shape) < len(prev_layer_output_shape) and \
-                self.get_type(layer) != 'Flatten':
+                self.get_type(layer) != 'Flatten' and \
+                self.get_type(previous_layers[0]) != 'InputLayer':
```
[the remark this answers is not in the document]
Comment (ej159) [reply snntoolbox/parsing/utils.py]: I can't remember exactly, best leave it out.

File: snntoolbox/simulation/target_simulators/pyNN_target_sim.py [hunk 2]
```diff
@@ -185,6 +216,11 @@ def simulate(self, **kwargs):
                  for amplitude in x_flat]
             self.layers[0].set(spike_times=spike_times)
 
+        from pynn_object_serialisation.functions import intercept_simulator
```
[the remark this answers is not in the document]
Comment (ej159) [reply snntoolbox/simulation/target_simulators/pyNN_target_sim.py]: This is a way of saving simulations such that they can be reloaded that @pabogdan developed. Currently I think this only works for SpiNNaker (Right @pabogdan?). I'll move this over to spinnaker_target_sim for now.

**Comment by ej159** [comment]
The SpiNNaker tests now work on my end. Thanks :)

**Pull request merged**